src/compiler/factory.ts

```diff
@@ -2798,6 +2798,20 @@ namespace ts {
         return tokenSourceMapRanges && tokenSourceMapRanges[token];
     }
 
+    export function getConstantValue(node: Node) {
+        const emitNode = node.emitNode;
+        if (emitNode && emitNode.flags & EmitFlags.ConstantValue) {

```

Comment:
It probably doesn't matter. Since constantValue is an expando property, I was trying to avoid some deoptimization cases.